knownvalue/list_elements.go

```diff
@@ -0,0 +1,57 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package knownvalue
+
+import (
+	"fmt"
+	"strconv"
+)
+
+var _ Check = ListElements{}
+
+// ListElements is a Check for asserting equality between the value supplied
+// to ListElementsExact and the value passed to the CheckValue method.
+type ListElements struct {
+	num int
+}
+
+// CheckValue verifies that the passed value is a list, map, object,
+// or set, and contains a matching number of elements.
+func (v ListElements) CheckValue(other any) error {
+	otherVal, ok := other.([]any)
+
+	if !ok {
+		return fmt.Errorf("expected []any value for ListElements check, got: %T", other)
+	}
+
+	if len(otherVal) != v.num {
+		expectedElements := "elements"
+		actualElements := "elements"
+
+		if v.num == 1 {
+			expectedElements = "element"
+		}
+
+		if len(otherVal) == 1 {
+			actualElements = "element"
+		}
+
+		return fmt.Errorf("expected %d %s for ListElements check, got %d %s", v.num, expectedElements, len(otherVal), actualElements)
+	}
+
+	return nil
+}
+
+// String returns the string representation of the value.
+func (v ListElements) String() string {
+	return strconv.FormatInt(int64(v.num), 10)
+}
+
+// ListElementsExact returns a Check for asserting that
+// a list num elements.
+func ListElementsExact(num int) ListElements {
```

Comment:
Totally happy to refactor to `<List|Map|Set>SizeExact`. Having some level of parity with the validators seems beneficial to me.

Only question I have if we go this route is the naming for the `object` equivalent - `ObjectNumAttributesExact`, `ObjectLenAttributesExact`, `ObjectAttributesCountExact`.........

Reply:
Ah, for _objects_ I would argue that its a little awkward to count their attributes since that's part of their static type definition. This differs from map elements where there could be an arbitrary amount from an API.

Reply:
Have renamed to  `<List|Map|Set>SizeExact`, and removed `ObjectAttributesExact`.